From 87c8eb386e20c00afb958db9aa6519f99566036a Mon Sep 17 00:00:00 2001
Message-Id: <87c8eb386e20c00afb958db9aa6519f99566036a.1425610844.git.chang-joon.lee@intel.com>
In-Reply-To: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
References: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Thu, 5 Feb 2015 19:07:48 +0530
Subject: [PATCH 02/27] MUST_REBASE [VPG]: drivers/video/adf: Export
 compute_m_n function

This patch exports the dp_pipe_compute_m_n function to be
used by eDP DRRS code to program M/N values for high/low
refresh rates.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5760
Change-Id: I3f9537f55117b2863a2b647b0eb057c36121b862
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 .../video/adf/intel/core/common/dp/gen_dp_pipe.c   |    2 +-
 .../adf/intel/include/core/common/dp/gen_dp_pipe.h |    3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c b/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c
index e785a5a..e8d52a5 100644
--- a/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c
+++ b/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c
@@ -38,7 +38,7 @@ static void compute_m_n(u32 m, u32 n,
 	}
 }
 
-static void dp_pipe_compute_m_n(u32 bits_per_pixel, u32 nlanes,
+void dp_pipe_compute_m_n(u32 bits_per_pixel, u32 nlanes,
 		u32 pixel_clock, u32 link_clock, struct intel_link_m_n *m_n)
 {
 	m_n->tu = 64;
diff --git a/drivers/video/adf/intel/include/core/common/dp/gen_dp_pipe.h b/drivers/video/adf/intel/include/core/common/dp/gen_dp_pipe.h
index 49dcae5..2bbd3e29 100644
--- a/drivers/video/adf/intel/include/core/common/dp/gen_dp_pipe.h
+++ b/drivers/video/adf/intel/include/core/common/dp/gen_dp_pipe.h
@@ -73,6 +73,9 @@ u32 dp_pipe_init(struct dp_pipe *pipe, struct device *dev,
 	struct intel_pipeline *pipeline, enum intel_pipe_type type);
 
 u32 dp_pipe_destroy(struct dp_pipe *pipe);
+void dp_pipe_compute_m_n(u32 bits_per_pixel, u32 nlanes,
+			u32 pixel_clock, u32 link_clock,
+			struct intel_link_m_n *m_n);
 
 int intel_adf_dp_hot_plug(struct dp_pipe *dp_pipe);
 int intel_dp_self_modeset(struct dp_pipe *dp_pipe);
-- 
1.7.9.5

